src/main/secret/secret_manager.cpp

```diff
@@ -108,7 +108,15 @@ unique_ptr<BaseSecret> SecretManager::DeserializeSecret(Deserializer &deserializ
 		    "Attempted to deserialize secret type '%s' which does not have a deserialization method", type);
 	}
 
-	return deserialized_type.deserializer(deserializer, {scope, type, provider, name});
+	auto function_entry = LookupFunctionInternal(type, provider);
+	if (!function_entry) {
+		throw InternalException(
+		    "Attempted to deserialize secret (type: '%s', provider: '%s') which does not have any functions registered",
+		    type, provider);
```

Comment:
I think this should not be an InternalException, given we might change the set of providers.

Reply:
This can't really be tested on a same version, but I think this has to be more of a IOException, where basically the secret can not be understood by DuckDB, but it's not a logical inconsistency AND it's recoverable